packages/iTwinUI-react/src/core/Table/Table.test.tsx

```diff
@@ -361,62 +361,100 @@ it('should handle checkbox clicks', async () => {
 it('should handle row clicks', async () => {
   const onSelect = jest.fn();
   const onRowClick = jest.fn();
+  const user = userEvent.setup();
+  const data = mockedData(5);
+
   const { container, getByText } = renderComponent({
     isSelectable: true,
     onSelect,
     onRowClick,
+    data,
   });
 
   expect(screen.queryByText('Header name')).toBeFalsy();
   const rows = container.querySelectorAll('.iui-table-body .iui-row');
-  expect(rows.length).toBe(3);
-
-  const user = userEvent.setup();
+  expect(rows.length).toBe(5);
 
-  // Shift click test #1
+  // Shift click special case test #1
   // By default, when no row is selected before shift click, start selecting from first row to clicked row
-  await user.keyboard('[ShiftLeft>]'); // Press Shift (without releasing it)
-  await user.click(getByText(mockedData()[1].name)); // [shiftKey: true]
+  await user.keyboard('[ShiftLeft>]'); // Hold Shift
+  await user.click(getByText(data[2].name));
 
   expect(rows[0].classList).toContain('iui-selected');
   expect(rows[1].classList).toContain('iui-selected');
-  expect(rows[2].classList).not.toContain('iui-selected');
+  expect(rows[2].classList).toContain('iui-selected');
+  expect(rows[3].classList).not.toContain('iui-selected');
   expect(onSelect).toHaveBeenCalledTimes(1);
   expect(onRowClick).toHaveBeenCalledTimes(1);
 
-  await userEvent.click(getByText(mockedData()[1].name)); // Deselect
+  await userEvent.click(getByText(data[1].name)); // Deselect; lastSelectedRowId = undefined -> 1
+
   expect(rows[0].classList).toContain('iui-selected');
   expect(rows[1].classList).not.toContain('iui-selected');
-  expect(rows[2].classList).not.toContain('iui-selected');
+  expect(rows[2].classList).toContain('iui-selected');
```

Comment:
If shift is still pressed here and you click on row 1, why is row 2 still selected? When I try this sequence in storybook I get different results - row 0 and 1 are selected and row 2 and 3 aren't.

Reply:
Ohh! That's true. Looks like `await user.keyboard('[ShiftLeft>]');` ([L380](https://github.com/iTwin/iTwinUI-react/pull/889/files#diff-c4aa401f671f853692462c6d1ee006969f80e3d02863b8a1a8919154aa3f67d9R380)) only holds shift for one click? But the [documentation](https://testing-library.com/docs/user-event/keyboard/#:~:text=Keys%20can%20be%20kept%20pressed%20by%20adding%20a%20%3E%20to%20the%20end%20of%20the%20descriptor.) says `>` should keep the key held continuously.

Or maybe I misunderstood. Maybe the docs means that if `>` is provided without a number at the end (e.g. `[ShiftLeft>5]`), then the key is only held for 1 click.

Do you know which is this true?

Reply:
I think the second thing. If you want a button held down for multiple keypresses then you need to add a number to signify how many.

Reply:
**[DONE]** Ohh, I see, thanks. To be safe, I explicitly released the shift before deselecting the row ([L390](https://github.com/iTwin/iTwinUI-react/pull/889/files#diff-c4aa401f671f853692462c6d1ee006969f80e3d02863b8a1a8919154aa3f67d9R390)).

Reply:
Also you changed this test from 'shift click 2 shift click 1' to 'shift click 2 click 1'

Reply:
> `{ }` should be used with 'Shift' instead of 'ShiftLeft' like `{Shift>}`

Yes, as per the documentation `{...}` is for [KeyboardEvent.key](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key) and `[...]` is for [KeyboardEvent.code](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code). I saw that other places in the repo were using the `{...}` version, so I changed all `[...]` examples to use that version too, for uniformity.

> Maybe # 1 is why the shift key wasn't being held forever?

I'm not sure, because like you mentioned, both versions don't hold shift continuously for me either.

> Also you changed this test from 'shiftclick 2 shiftclick 1' to 'shiftclick 2 click 1'

I think both the old and current versions are actually doing 'shiftclick 2 click 1'. The old version doesn't have the explicit release shift key like on [L390](https://github.com/iTwin/iTwinUI-react/pull/889/files#diff-c4aa401f671f853692462c6d1ee006969f80e3d02863b8a1a8919154aa3f67d9R390), but since `>` on [L378](https://github.com/iTwin/iTwinUI-react/pull/889/files#diff-c4aa401f671f853692462c6d1ee006969f80e3d02863b8a1a8919154aa3f67d9L378) was not holding shift for multiple click like discussed above, that click was also actually a non-shift click.

Or maybe I misunderstood what you meant?

Reply:
> Yes, as per the documentation `{...}` is for [KeyboardEvent.key](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key) and `[...]` is for [KeyboardEvent.code](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code). I saw that other places in the repo were using the `{...}` version, so I changed all `[...]` examples to use that version too, for uniformity.

Ah, I see

> I'm not sure, because like you mentioned, both versions don't hold shift continuously for me either.

Well that's unfortunate.

> I think both the old and current versions are actually doing 'shiftclick 2 click 1'. The old version doesn't have the explicit release shift key like on [L390](https://github.com/iTwin/iTwinUI-react/pull/889/files#diff-c4aa401f671f853692462c6d1ee006969f80e3d02863b8a1a8919154aa3f67d9R390), but since `>` on [L378](https://github.com/iTwin/iTwinUI-react/pull/889/files#diff-c4aa401f671f853692462c6d1ee006969f80e3d02863b8a1a8919154aa3f67d9L378) was not holding shift for multiple click like discussed above, that click was also actually a non-shift click.

> Or maybe I misunderstood what you meant?

It looked like you initially meant for the test to be 'shiftclick 2 shiftclick 1'. If you meant for it to be 'shiftclick 2 click 1' then that's fine. But if you meant for it to be 'sc2 sc1' then you should change it to run like that.

Reply:
Ohh, thanks for pointing that out. I guess I was just testing `sc c` and not `sc sc` in both cases. So I think it's alright to leave it as it.